core/src/sql/value/idiom_recursion.rs

```diff
@@ -0,0 +1,158 @@
+use crate::{
+	cnf::IDIOM_RECURSION_LIMIT,
+	ctx::Context,
+	dbs::Options,
+	doc::CursorDoc,
+	err::Error,
+	sql::{part::RecursionPlan, Array, Part},
+};
+
+use super::Value;
+use reblessive::tree::Stk;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Recursion<'a> {
+	pub min: &'a u32,
+	pub max: Option<&'a u32>,
+	pub iterated: &'a u32,
```

Comment:
These fields are probably fine being copied as `size_of::<&u32>() >= size_of::<u32>()` on most platforms. Also simplifies the type a bit.

Reply:
I mean, does it actually make it much easier? We need the lifetime anyway for the other borrowed fields

Reply:
Fixed in [39d5e84](https://github.com/surrealdb/surrealdb/pull/5086/commits/39d5e84777dda081de1d4dce4470a8e3a481ea88)